apps/aecore/src/aec_pow_cuckoo.erl

```diff
@@ -620,15 +621,22 @@ exec_run(Cmd, Dir, Args) ->
     PortName = {spawn_executable, os:find_executable(Cmd, Dir)},
     try
         Port = erlang:open_port(PortName, PortSettings),
-        {os_pid, OsPid} = erlang:port_info(Port, os_pid),
-        ?debug("External mining process started with OS pid ~p", [OsPid]),
-        {ok, Port, OsPid}
+        case erlang:port_info(Port, os_pid) of
+            {os_pid, OsPid} ->
+                ?debug("External mining process started with OS pid ~p", [OsPid]),
+                {ok, Port, OsPid};
+            undefined ->
+                ?error("External mining process finished before ~p could acquire the OS pid", [?MODULE]),
```

Comment:
It feels like this shall be a warning rather than an error?
If this case happens, I expect no OS process leaked and the output of the OS process (e.g. if exited because extremely quickly to start and find solution) parsed in the main loop `wait_for_result`.

Also, for testing this error would still trigger test failure for [this check on logs](https://github.com/aeternity/epoch/blob/a9e8f8ae8f7dc3e6dcf2da30342684ae37218661/system_test/helpers/aest_nodes.erl#L94).

Reply:
Changed to warning.